### DISCUSSION TITLE
Replace Checkstyle & Spring Javaformat with Spotless

### DISCUSSION THREAD

**Discussion by maciejwalkowiak** [awspring/spring-cloud-aws#32] [General]
Checkstyle significantly raises the bar for external contributors, as build is failing if code does not match Checkstyle constraints but it's not trivial to configure an IDE to conform to Checkstyle rules used in Spring Cloud AWS.

I suggest we switch to alternative - [Spotless](https://github.com/diffplug/spotless/). Huge Spotless advantage is that in addition to `check` goal it also has `apply` goal that automatically fixes all the style & formatting issues.

If you want to try it out I can quickly prepare a PR cc @eddumelendez @MatejNedic

**Comment by MatejNedic** [comment]
Agree we can check it out. Setting current checkstyle in IDE is not trivial indeed. If we can have something faster for set up, I think we should go for it.

**Reply by arunkpatra** [reply]
Tell me about it :-) It scared me to death.

**Comment by mkatircioglu** [comment]
Wouldn't it better to be compatible with what Spring Boot & Spring Cloud are actively using? @maciejwalkowiak

**Reply by maciejwalkowiak** [reply]
Afaik there is no 100% alignment between spring projects and while it probably works well for long time contributors, currently the initial setup is just a bit of hassle for someone who just wants to submit a PR

**Comment by eddumelendez** [comment]
As far as do not apply changes to most of the current codebase I think it will be ok

**Reply by eddumelendez** [reply]
For spring format we can add `<goal>apply<goal>` [here](https://github.com/awspring/spring-cloud-aws/blob/ef18e6cfe7dc0982fe2a655fa78e5a476efa2406/pom.xml#L173)

**Comment by maciejwalkowiak** [comment]
https://github.com/awspring/spring-cloud-aws/pull/281
